### DISCUSSION TITLE
LSP keymaps and syntax highlighting

### DISCUSSION THREAD

**Discussion by jamesredwards** [mrcjkb/rustaceanvim#430] [Q&A]
Hi, I'm really struggling to get the out of the box experience with this plugin. When trying to use the same keymaps as lspconfig in nvchad i can get that working, but the syntax highlighting then breaks. I appreciate syntax highlighting is nothing to do with rustaceanvim but clearly I'm doing something wrong with the setup. Here is the code I am using:

```lua
  {
    "mrcjkb/rustaceanvim",
    version = "^4", -- Recommended
    ft = { "rust" },
    config = function(_, _)
      local on_attach = require("nvchad.configs.lspconfig").on_attach
      local on_init = require("nvchad.configs.lspconfig").on_init
      local capabilities = require("nvchad.configs.lspconfig").capabilities
      vim.g.rustaceanvim = {
        server = {
          on_attach = function(client, bufnr)
            vim.lsp.inlay_hint.enable(true, { bufnr = bufnr })
            on_attach(client, bufnr)
          end,
          on_init = on_init,
          capabilities = capabilities,
        },
      }
    end,
    lazy = false,
  },
I've searched elsewhere but can't seem to figure out what the problem is. Any guidance much appreciated!

**Comment by mrcjkb** [comment]
Hey :wave: 

Since this has to do with nvchad, I'm marking this as a duplicate of https://github.com/mrcjkb/rustaceanvim/discussions/351#discussioncomment-9097402
